python_modules/dagster/dagster/_core/definitions/asset_layer.py

```diff
@@ -410,6 +410,7 @@ def from_graph_and_assets_node_mapping(
         asset_checks_defs_by_node_handle: Mapping[NodeHandle, "AssetChecksDefinition"],
         observable_source_assets_by_node_handle: Mapping[NodeHandle, "SourceAsset"],
         source_assets: Sequence["SourceAsset"],
+        loadable_assets: Sequence[Union["AssetsDefinition", "SourceAsset"]],
```

Comment:
I'm surprised that anything new would be needed here, given that we already need to access this for loading inputs

Reply:
We discard the ones that aren't needed for loading inputs

Reply:
Where does that happen? My thought would be that it would be more consistent to handle this there.

Reply:
Implicitly in this function, if you look at the other arguments, all the attached relevant assets here have an associated node handle, or are a source asset used for loading. In our case, there might not be a relevant node handle, and so I think a separate argument is required.